server/src/test/java/org/elasticsearch/search/query/SearchQueryIT.java

```diff
@@ -1568,6 +1568,15 @@ public void testRangeQueryWithTimeZone() throws Exception {
         assertThat(searchResponse.getHits().getAt(0).getId(), is("4"));
     }
 
+    /**
+     * Test range with a custom locale, e.g. "de" in this case. Documents here mention the day of week
+     * as "Mi" for "Mittwoch (Wednesday" and "Do" for "Donnerstag (Thursday)" and the month in the query
+     * as "Dez" for "Dezember (December)".
+     * Note: this test currently needs the JVM arg `-Djava.locale.providers=SPI,COMPAT` to be set.
```

Comment:
Could we assert that this arg is set like this in this test? Just trying to think of a way to prevent this comment from living forever 😕

Reply:
Good point, and fail with a better explanation if not set? As far as I understood @pgomulka, the need for this argument will stay, but we might be able to automate setting it through the IDE as well with issues like #46118? In that case I don't know how we can detect the point where we can remove the comment here, since the argument will still be set, only not manually but through some gradle setup magic?

Reply:
agree that assert at the beginning of the test would be great. I think we will need SPI and COMPAT options to be set unless there will be changes in JDK that would this redundant (like defaulting Week related options to ISO standard). 
maybe we can just leave this as a TODO and this will get attention at some point?